src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1267,6 +1267,20 @@ AddRteSubqueryToAttributeEquivalenceClass(AttributeEquivalenceClass
 										  PlannerInfo *root,
 										  Var *varToBeAdded)
 {
+#if PG_VERSION_NUM >= PG_VERSION_16
+	if (root->simple_rel_array_size <= varToBeAdded->varno)
```

Comment:
can  we create a function, like `IsVarRelOptOuterJoin(root, varToBeAdded)`, and then move all the ifdefs there?